src/main/java/com/dungeoncrawler/Javiarenka/shop/Shop.java

```diff
@@ -0,0 +1,58 @@
+package com.dungeoncrawler.Javiarenka.shop;
+
+import com.dungeoncrawler.Javiarenka.character.Hero;
+import com.dungeoncrawler.Javiarenka.character.HeroClass;
+import com.dungeoncrawler.Javiarenka.equipment.Armor;
+import com.dungeoncrawler.Javiarenka.equipment.Weapon;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Shop {
+
+    private static List<Armor> allAvailableArmor = new ArrayList<>(Arrays.asList(null, null, null, null, null)); // todo
+    private static List<Weapon> allAvailableWeapon = new ArrayList<>(Arrays.asList(null, null, null, null, null)); // todo
+
+    public static List<Armor> getNewArmor(Object object) {
+        Hero hero = validation(object);
+        return getAvailableArmour(getHeroClass(hero));
+    }
+
+    public static List<Weapon> getNewWeapon(Object object) {
+        Hero hero = validation(object);
+        return getAvailableWeapon(getHeroClass(hero));
+    }
+
+    private static List<Weapon> getAvailableWeapon(HeroClass heroClass) {
+        return allAvailableWeapon
+                .stream()
+                .filter((Weapon weapon) -> weapon.getClassRestrictions().contains(heroClass))
+                .collect(Collectors.toList());
+    }
+
+    private static List<Armor> getAvailableArmour(HeroClass heroClass) {
```

Comment:
```getAvailableArmour```
zmieniłbym na 
```getAvailableArmor```
to nie błąd, bo jedna i druga pisownia jest poprawna, ale trzymajmy się tej konwencji, która już istnieje

Reply:
Ponieważ z jakiegoś powodu przy pisaniu docelowej metody getNewEquipment() kod mi się nie kompilował, rozbiłam ją na getNewArmor() i getNewWeapon() - w trakcie siedzenia nad kodem (nieświadomie, powiedzmy sobie szczerze XD) rozwiązałam problem i wróciłam do pierwotnego pomysłu - jeżeli jednak potrzebowalibyśmy w/w metod dodam je do klasy